lib/notifications/display.dart

```diff
@@ -456,36 +456,58 @@ class NotificationDisplayManager {
 
   static String _personKey(Uri realmUrl, int userId) => "$realmUrl|$userId";
 
+  /// Provides the route and the account ID by parsing the notification URL.
+  ///
+  /// The URL must have been generated using [NotificationOpenPayload.buildUrl]
+  /// while creating the notification.
+  ///
+  /// Returns null if the associated account is not found in the global
+  /// store.
+  static ({Route<void> route, int accountId})? routeForNotification({
+    required GlobalStore globalStore,
+    required Uri url,
+  }) {
+    assert(debugLog('got notif: url: $url'));
+    assert(url.scheme == 'zulip' && url.host == 'notification');
+    final payload = NotificationOpenPayload.parseUrl(url);
+
+    final account = globalStore.accounts.firstWhereOrNull((account) =>
+      account.realmUrl == payload.realmUrl && account.userId == payload.userId);
```

Comment:
How likely do we think it is that these realm URLs could differ in boring ways that shouldn't matter? Like `https://chat.zulip.org/` vs. ``https://chat.zulip.org`. Do we need to just compare `.origin` instead, like we do for most (all?) other realm-URL checks?

Reply:
Right, I've added a commit that changes this query to compare only using `.origin`.